alkira/resource_alkira_checkpoint.go

```diff
@@ -0,0 +1,326 @@
+								"management server. If you choose to use manual configuration " +
+								"Alkira will provide the customer information about the checkpoint " +
+								"instances so that you can manually configure the firewall.",
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"MANUAL", "AUTOMATED"}, false),
+						},
+						"credential_id": {
+							Description: "The credential id of the Checkpoint Firewall's management server. ",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"domain": {
+							Description: "Management server domain.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"global_cidr_list_id": {
+							Description: "The id of the global cidr list to be associated with " +
+								"the management server.",
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"ips": {
+							Description: "Management server ips.",
+							Type:        schema.TypeList,
+							Required:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						"reachability": {
+							Description: "This option specifies whether the management server is " +
+								"publicly reachable or not. If the reachability is private then you " +
+								"need to provide the segment to be used to access the management server. ",
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"segment_id": {
+							Description: "The id of the segment to be used to access the management server.",
+							Type:        schema.TypeInt,
+							Optional:    true,
+						},
+						"type": {
+							Description: "The type of the management server.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"user_name": {
+							Description: "The user_name of the management server.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+					},
+				},
+			},
+			"max_instance_count": {
+				Description: "The maximum number of Checkpoint Firewall instances that should be " +
+					"deployed when auto-scale is enabled. Note that auto-scale is not supported " +
+					"with Checkpoint at this time. max_instance_count must be greater than or " +
+					"equal to min_instance_count.",
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"min_instance_count": {
+				Description: "The minimum number of Checkpoint Firewall instances that should be " +
+					"deployed at any point in time.",
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+			},
+			"name": {
+				Description: "Name of the Checkpoint Firewall service.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"pdp_ips": {
+				Description: "The IPs of the PDP Brokers.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"segment_names": {
```

Comment:
Hmmm, there is segment_id already... Is this duplicated? And it should be `segment_id` to be consistent.